lib/doorkeeper/oauth/refresh_token_request.rb

```diff
@@ -23,10 +23,8 @@ def initialize(server, refresh_token, credentials, parameters = {})
         @original_scopes = parameters[:scope] || parameters[:scopes]
         @refresh_token_parameter = parameters[:refresh_token]
 
-        if credentials
-          @client = Application.by_uid_and_secret credentials.uid,
-                                                  credentials.secret
-        end
+        @client = Application.by_uid_and_secret(credentials.uid,
+                                                credentials.secret) if credentials
```

Comment:
This is a surprising `if` back and down there. I would leave the multi-line `if`.